### PR TITLE
Tech 1218 sla

### PR DESCRIPTION
**Pull request by gabezurita**
Hi @joshsz,

Would you mind checking this PR for me?

Tech Support is switching to a new Jira workflow, one which has new "closed" statuses. The jira_reporting gem needs a code change to account for these statuses. There is no rush on this, as we can manually fix any issues that aren't counted as closed on time.

In addition to accounting for the new TS workflow 5.0, this pull request does the following:
- Adds a config file to the Holidays gem to reflect Optoro's holidays.
- Allows users to manually set SLA due time fields, and then calculates whether an issue is late based on that manually set value. In the past, the program was strictly calculating whether an issue was late or not based off of the issue's priority level.

### CODE REVIEW

File: bin/jira_sla_update [hunk 1]
```diff
@@ -32,11 +32,18 @@ query = %Q{
...
+
+update_start_time = Time.now
+
+puts "The time is #{update_start_time}. There are #{total_ts_issues} issues in the query. Each issue takes about five seconds to update. The estimated amount of time this update will take is #{total_ts_issues*5/60} minutes to complete -- expected end time: #{update_start_time+total_ts_issues*5}"
+
+
 ts_issues.each do |issue|
   issue.set_sla_due_time if issue.sla_due_time == nil
-  issue.set_over_sla if issue.sla_time_ratio > 1 && issue.over_sla == nil
+  issue.set_over_sla if issue.over_sla == nil
```
Comment (joshsz) [inline bin/jira_sla_update]: if `set_over_sla` transmits data back to JIRA you probably want to name it with a `!`, so `set_over_sla!`. That's generally ruby idiom.

File: lib/jira_issue.rb [hunk 2]
```diff
@@ -3,6 +3,8 @@
 class JiraIssue
   # This variable exists for tickets created before 2015-03-06 15:55:16 -0500. Not used for any tickets after that.
   FIVE_P_DATE = Time.at(1425675316)
+  YES = "11901"
```
Comment (joshsz) [inline lib/jira_issue.rb]: `YES` and `NO` are ambiguous, consider naming them to mean something more specific

File: lib/jira_issue.rb [hunk 3]
```diff
@@ -55,10 +57,10 @@ def set_sla_due_time(due_time = nil)
   end
 
   def set_over_sla(over = nil)
-    over ||= calculate_over_sla
+    over ||= is_over_sla?
     return if over.nil?
     jira_issue = jiralicious_issue
-    jira_issue.fields.set_id("customfield_12901", over ? "11901" : "11900")
+    jira_issue.fields.set_id("customfield_12901", YES)
```
Comment (joshsz) [inline lib/jira_issue.rb]: consider introducing a constant for `"customfield_12901"` and any other `"customfield"`s

**Comment by gabezurita** [comment]
@joshsz, I've made the requested changes -- thanks very much for the input! Please let me know if there's anything else I should do prior to merge.

File: bin/jira_sla_update (hunk 1 above)
Comment (gabezurita) [inline bin/jira_sla_update]: Agreed. Thanks for the reminder! I'll make this change.

File: lib/jira_issue.rb (hunk 2 above)
Comment (gabezurita) [inline lib/jira_issue.rb]: This is true. I've gone ahead and made YES and NO more specific: OVER_SLA_FIELD_YES & OVER_SLA_FIELD_NO

File: lib/jira_issue.rb (hunk 3 above)
Comment (gabezurita) [inline lib/jira_issue.rb]: Sorry for being lazy and not doing that ahead of time -- I went ahead and introduced constants for all vague Jira customfields.

**Comment by joshsz** [comment]
@gabezurita looks awesome! merged :)

**Pull request merged**